sunpy/net/dataretriever/sources/adapt.py

```diff
@@ -0,0 +1,96 @@
+from sunpy.net import attrs
+from sunpy.net.fido_factory import Fido
+from sunpy.net.dataretriever import GenericClient
+from sunpy.net.dataretriever.attrs.adapt import *
+import sunpy.net.dataretriever.attrs as atrs
+from sunpy.net.attr import SimpleAttr
+import astropy.units as u
+import sunpy.coordinates
+
+__all__ = ['ADAPTClient']
+
+
+class ADAPTClient(GenericClient):
+    """
+    Provides access to the ADvanced Adaptive Prediction Technique (ADAPT) products of the National Solar Observatory.
+
+        Searches data hosted by the `National Solar Observatory <https://gong.nso.edu/adapt/maps/gong/>`__
+
+        Examples
+        --------
+        >>> from sunpy.net import Fido, attrs as a
+        >>> get_date, get_date_end = adapt_source.carrington_time(CR, frames)
+        >>> LngType = '0' # 0 is carrington, 1 is central meridian
+        >>> print (f"Searching for images from {get_date} to {get_date_end}\n")
+        >>> res = Fido.search(attrs.Instrument('adapt'), attrs.Time(get_date, get_date_end), ADAPTLngType(LngType))
+        >>> print(res)
+        >>> ret =  Fido.fetch(res, path=make_dir())
+
+
+    Searching for images from 2017-07-20T06:03:05 to 2017-07-20T12:03:05
+
+    Results from 1 Provider:
+
+    1 Results from the ADAPTClient:
+
+        Start Time               End Time        ... minutes_since_last_obs seconds_since_last_obs
+    ----------------------- ----------------------- ... ---------------------- ----------------------
+    2017-07-20 08:00:00.000 2017-07-20 08:00:59.999 ...                     56                      0
+
+
+    Fetching search_result:
+    Files Downloaded: 100%|███████████████████████████████████████████| 1/1 [00:00<00:00,  7.65file/s]
+    Success!
```

Comment:
```suggestion
```
This will need to be replaced with just the output from the search block.
We want to avoid downloading files where possible to reduce test time and server load on providers.